src/inject/dynamic-theme/style-manager.ts

```diff
@@ -465,20 +476,23 @@ export function manageStyle(element: StyleElement, {update, loadingStart, loadin
     };
 }
 
-async function linkLoading(link: HTMLLinkElement) {
+async function linkLoading(link: HTMLLinkElement, loadingID: number) {
     return new Promise<void>((resolve, reject) => {
         const cleanUp = () => {
             link.removeEventListener('load', onLoad);
             link.removeEventListener('error', onError);
+            loadingLinkStyles.delete(loadingID);
         };
         const onLoad = () => {
             cleanUp();
+            logInfo(`Linkelement ${loadingID} has been loaded`);
             resolve();
         };
         const onError = () => {
             cleanUp();
-            reject(`Link loading failed ${link.href}`);
+            reject(`Linkelement ${loadingID} couldn't be loaded. ${link.href}`);
         };
+        loadingLinkStyles.set(loadingID, reject);
```

Comment:
As far as you put rejectors here, give `loadingLinkStyles` a proper name. Also, take a look if you can use `link` element itself as a key in a WeakSet, Set, WeakMap or Map instead of numeric IDs.